chainstate/src/detail/tests/test_framework.rs

```diff
@@ -60,12 +60,16 @@ impl BlockTestFramework {
         }
     }
 
-    pub fn new() -> Self {
+    pub(crate) fn new() -> Self {
         let chainstate = setup_chainstate();
         Self::with_chainstate(chainstate)
     }
 
-    pub fn random_block(&self, parent_block: &Block, params: Option<&[TestBlockParams]>) -> Block {
+    pub(crate) fn random_block(
+        &self,
+        parent_block: &Block,
+        params: Option<&[TestBlockParams]>,
+    ) -> Block {
```

Comment:
I think the seedable generator should be passed to this function as well (and related) and used to generate the block contents. Otherwise we will generate chains with deterministic lengths but the block contents will still not be reproducible.

You may want to do that after re-basing on top of #299, there will be conflicts.

Reply:
Exactly. Every test should use the rng exactly once. Then, wherever the rng is needed, should be passed there as a mutable reference.